src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java

```diff
@@ -359,12 +359,19 @@ public List<String> getSecurityRoles(String user) {
 
             // Security roles should only contain roles that exist in the roles dynamic config.
             // We should filter out any roles that have hidden rolesmapping.
-            return tmp == null
-                ? ImmutableList.of()
-                : tmp.getOpendistro_security_roles()
+            if (tmp == null) {
+                return ImmutableList.of();
+            } else if (!tmp.getDirect_security_roles().isEmpty()) {
```

Comment:
This will silently ignore  the old attribute - even if still exists. This can lead to confusion.

Reply:
will add a log here